File: Data Structures & Algorithms/two-integer-sum/submission-5.go
func twoSum(nums []int, target int) []int {
	m := make(map[int]int)
	for ind, val := range nums{
		t := target - val
		if v, ok := m[t]; ok && ind != v{
			return []int{v, ind}
		}
		m[val] = ind
	}
	return []int{1, 2}
}

File: Data Structures & Algorithms/two-integer-sum-ii/submission-5.py
class Solution:
    def twoSum(self, numbers: List[int], target: int) -> List[int]:
        l = 0
        r = len(numbers) - 1

        while l < r:
            t = numbers[l] + numbers[r]
            if t == target:
                return [l+1,r+1]
            if t < target:
                l += 1
            else:
                r -= 1
        
        return []


        